src/cache_filesystem.cpp

```diff
@@ -361,17 +361,32 @@ unique_ptr<FileHandle> CacheFileSystem::GetOrCreateFileHandleForRead(const strin
 	unique_ptr<FileHandle> file_handle = nullptr;
 	{
 		const auto latency_guard = profile_collector->RecordOperationStart(IoOperation::kOpen);
-		file_handle = internal_filesystem->OpenFile(path, flags | FileOpenFlags::FILE_FLAGS_PARALLEL_ACCESS, opener);
+		file_handle = internal_filesystem->OpenFile(file, flags | FileOpenFlags::FILE_FLAGS_PARALLEL_ACCESS, opener);
 	}
 	DUCKDB_LOG_OPEN_CACHE_MISS_PTR((file_handle));
 	return CreateCacheFileHandleForRead(std::move(file_handle));
 }
 
+// LCOV_EXCL_START
+unique_ptr<FileHandle> CacheFileSystem::OpenFileExtended(const OpenFileInfo &file, FileOpenFlags flags,
+                                                         optional_ptr<FileOpener> opener) {
+	InitializeGlobalConfig(opener);
+	if (flags.OpenForReading()) {
+		return GetOrCreateFileHandleForRead(file, flags, opener);
+	}
+
+	// Otherwise, we do nothing (i.e. profiling) but wrapping it with cache file handle wrapper.
+	// For write handles, we still need the connection_id for consistency
+	auto file_handle = internal_filesystem->OpenFile(file, flags, opener);
+	return make_uniq<CacheFileSystemHandle>(std::move(file_handle), *this,
+	                                        /*dtor_callback=*/[](CacheFileSystemHandle & /*unused*/) {});
+}
+
 unique_ptr<FileHandle> CacheFileSystem::OpenFile(const string &path, FileOpenFlags flags,
                                                  optional_ptr<FileOpener> opener) {
 	InitializeGlobalConfig(opener);
 	if (flags.OpenForReading()) {
-		return GetOrCreateFileHandleForRead(path, flags, opener);
+		return GetOrCreateFileHandleForRead(OpenFileInfo(path), flags, opener);
```

Comment:
Now `OpenFile` and `OpenFileInfo` look almost the same, shall we combine them to the same implementation?
```c++
unique_ptr<Handle> OpenFile(path) {
  return OpenFileExtended(OpenInfo(path));
} 
```